onnxruntime/core/providers/cuda/tensor/gather_elements_impl.cu

```diff
@@ -98,6 +98,34 @@ struct FuncAssignment {
   __device__ __inline__ void operator()(T* start_addr, size_t index, T value) const { start_addr[index] = value; }
 };
 
+template <class T>
```

Comment:
I was looking for a line initializing the memory to 0 (or a suitable value depending on the reduction) but I did not find it.